core/src/operator.rs

```diff
@@ -456,29 +478,31 @@ where
         // call withdrawFillers(withdrawal_idx) check the returned id is our operator id.
         // calculate the move_txid, txIdToDepositId(move_txid) check the returned id is withdrawal_idx
         if let Some(citrea_client) = &self.citrea_client {
+            // See: https://gist.github.com/okkothejawa/a9379b02a16dada07a2b85cbbd3c1e80
             let params = rpc_params![
                 json!({
                     "to": "0x3100000000000000000000000000000000000002",
-                    "data": format!("0xc045577b00000000000000000000000000000000000000000000000000000000{}", hex::encode(withdrawal_idx.to_be_bytes())), // See: https://gist.github.com/okkothejawa/a9379b02a16dada07a2b85cbbd3c1e80
+                    "data": format!("0xc045577b00000000000000000000000000000000000000000000000000000000{}",
+                    hex::encode(withdrawal_idx.to_be_bytes())),
                 }),
                 "latest"
             ];
             let response: String = citrea_client.request("eth_call", params).await?;
-            let operator_idx_response = &response[58..66];
-            let operator_idx_as_vec = hex::decode(operator_idx_response).unwrap();
+
+            let operator_idx_as_vec = hex::decode(&response[58..66]).unwrap();
             let operator_idx = u32::from_be_bytes(operator_idx_as_vec.try_into().unwrap());
+
             if operator_idx - 1 != self.idx as u32 {
                 return Err(BridgeError::InvalidOperatorIndex(
                     operator_idx as usize,
                     self.idx,
                 ));
             }
 
-            // calculate move_txid
-
+            // Calculate move_txid.
             let move_tx_handler = transaction_builder::create_move_tx(
                 deposit_outpoint,
-                &EVMAddress([0u8; 20]),
+                &EVMAddress([0u8; 20]), // TODO: This doesn't look OK.
```

Comment:
This is ok, this is only for calculating the move_txid, that EVMaddress is written to the witness so does not effect the txid. Maybe we should note this here.

Reply:
Or have a different function to calculte just the txid, since we use this several times